packages/scratch-vm/src/typescript-support/Extension.ts

```diff
@@ -107,6 +133,13 @@ export abstract class Extension
         return !!value;
       case ArgumentType.Note:
         return parseInt(value);
+      case ArgumentType.Angle:
+        return parseInt(value);
+      case ArgumentType.Matrix:
+        console.log(this.toMatrix(`${value}`))
+        return this.toMatrix(`${value}`);
+      case ArgumentType.Color:
+        return Cast.toRgbColorObject(value);
```

Comment:
note that `Cast.toRgbColorObject` is a `JS` function, which means it returns `any`, making the return type of `castToType` `any`. Due to this, I think it might be worth defining a `TS` equivalent to `RGBObject`/ `Cast.toRgbColorObject` (or maybe even `Cast` itself).